frappe/utils/data.py

```diff
@@ -635,39 +637,61 @@ def format_datetime(datetime_string: DateTimeLikeObject, format_string: str | No
 	return formatted_datetime
 
 
-def format_duration(seconds, hide_days=False):
+DURATION_DATA = {
+	"years": {"factor": 12 * 1000000000, "abbr": "y"},
+	"months": {"factor": 1000000000, "abbr": "mo"},
+	"weeks": {"factor": 7 * 24 * 60 * 60, "abbr": "w"},
+	"days": {"factor": 24 * 60 * 60, "abbr": "d"},
+	"hours": {"factor": 60 * 60, "abbr": "h"},
+	"minutes": {"factor": 60, "abbr": "min"},
+	"seconds": {"factor": 1, "abbr": "s"},
+}
+
+
+def format_duration(seconds: int, duration_options: dict[str, bool] | None = None) -> str:
 	"""Converts the given duration value in float(seconds) to duration format
 
 	example: converts 12885 to '3h 34m 45s' where 12885 = seconds in float
 	"""
 
+	# Be tolerant, allow set instead of dict, and bool for BC.
+	if isinstance(duration_options, set):
+		duration_options = {option: True for option in duration_options}
+	elif isinstance(duration_options, bool):
+		duration_options = {"hide_days": duration_options}
+
 	seconds = cint(seconds)
+	if not seconds:
+		return ""
+
+	components = []
+	duration = seconds_to_duration(seconds, duration_options)
```

Comment:
Like in JS, the actual logic was split out to the new `seconds_to_duration()`, see below.